cmd/kots/cli/reset-tls.go

```diff
@@ -0,0 +1,148 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ResetTLSCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "reset-tls",
+		Short:         "Reverts kurl_proxy to a self-signed TLS certificate",
+		Long:          ``,
+		SilenceUsage:  true,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			v := viper.GetViper()
+
+			namespace := v.GetString("namespace")
+			if len(args) == 1 {
+				namespace = args[0]
+			} else if len(args) > 1 {
+				fmt.Printf("more than one argument supplied: %+v\n", args)
+				os.Exit(1)
+			}
+
+			if namespace == "" {
+				fmt.Printf("a namespace must be provided as an argument or via -n/--namespace\n")
+				os.Exit(1)
+			}
+
+			err := deleteKotsTLSSecret(namespace)
+			if err != nil {
+				return err
+			}
+
+			err = resetKurlProxyPod(namespace)
+			if err != nil {
+				return err
+			}
+
+			err = checkTLSSecret(namespace, time.Second*10)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringP("namespace", "n", "default", "namespace in which kots/kotsadm is installed")
+
+	return cmd
+}
+
+func deleteKotsTLSSecret(namespace string) error {
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return errors.Wrap(err, "failed to create k8s client")
+	}
+
+	existingSecret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), "kotsadm-tls", metav1.GetOptions{})
+	if err != nil {
+		if !kuberneteserrors.IsNotFound(err) {
+			return errors.Wrap(err, "failed to lookup secret")
+		}
+	}
+
+	if existingSecret.Name != "" {
+		err = clientset.CoreV1().Secrets(namespace).Delete(context.TODO(), existingSecret.Name, metav1.DeleteOptions{})
```

Comment:
This secret is mounted in kotsadm.  Does this need additional handling?

https://github.com/replicatedhq/kURL/blob/master/addons/kotsadm/1.41.1/kotsadm.yaml#L71-L73

Reply:
Shouldn't need anything, no.  Deleting the secret doesn't impact the pod in any way, the pod still has the secret mounted and nothing changes until the pod is redeployed.

Reply:
Shouldn't the pod have the new secret?

Reply:
If kurl_proxy boots up and the secret doesn't exist, it creates the self-signed cert and saves it as the kotsadm-tls secret.  All we have to do here is delete the previous secret to remove any problems and redeploy kurl_proxy.

Reply:
References to this in the kots deployment appear to be old artifacts.  They have been removed in this PR.